### ISSUE TITLE
[Flatpak] Refresh button for the hardware key is ineffective

### ISSUE THREAD

**Issue by Freeedim** [keepassxreboot/keepassxc#8213]
I use the graphical KDE Plasma menu to launch KeePassXC 2.7.1, which I installed with KDE Discover (the Flatpak packaging).

Edit:
~~If I launch KeePassXC without my YubiKey 5 being plugged, the "Hardware key" dropdown list and the "Refresh" button are is greyed and disabled. If I then plug the YubiKey nothing will get enabled and clicking on the "Refresh" button will have no effect.~~

If I launch KeePassXC without my YubiKey 5 being plugged, the "Hardware key" dropdown list  is greyed and disabled. If I then plug the YubiKey then click on the "Refresh" button, nothing will happen, i.e. the key won't be detected and the dropdown list will remain greyed and disabled.

In that situation, the only solution I could find was to shut down KeePassXC, plug my Yubikey and re-launch KeePassXC.

~~Maybe the only problem is having the "Refresh" button disabled when there is no hardware key detected at startup.~~

**Comment by droidmonkey** [comment]
That's odd, the button is only disabled when looking for hardware keys. This means that never finishes correctly in flatpak. Will need to look into that or add a timeout.

**Comment by Freeedim** [comment]
Actually I made a mistake (I was in the sun light with a dark theme...)

The "Refresh" button is **not** disabled. But it has just no effect, i.e. the "hot-plugged" key is never found, forcing to shutdown KeePassXC and re-launch it with the YubiKey already plugged.

**Comment by droidmonkey** [comment]
This seems like a flatpak issue that is not detecting when new devices are plugged in. We use libusb to poll the connected devices on demand. If flatpak isolation is not presenting an updated list of devices then we'll never see them.

**Comment by Freeedim** [comment]
Might be related to:
https://github.com/flatpak/flatpak/issues/4723

**Comment by louib** [comment]
I'm also facing this problem with the Flatpak version.

**Comment by the-wolfman** [comment]
> This seems like a flatpak issue that is not detecting when new devices are plugged in. We use libusb to poll the connected devices on demand. If flatpak isolation is not presenting an updated list of devices then we'll never see them.

I agree. Pulling the HW key or locking the screen (e.g. screensaver) while KeepassXC is running seems to be droping the item from the list, but "re-pluging" does not populate that list again. Can be a real pain if you did not save your latest changes. Only option in that case afaik is to temporarily store elsewhere with different security settings.
